### PR TITLE
Added #14426: Makes all Manufacturer links dynamic, not just warranty lookup

### PR DESCRIPTION
**Pull request by DrekiDegga**
# Description

This makes all manufacturer links dynamic just like warranty lookup.

Added #14426 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

I've tested working on latest from my test environment.

**Test Configuration**:
* PHP version: 7.4.33
* MySQL version 10.11.4-MariaDB-1~deb12u1 Debian 12
* Webserver version apache2 2.4.56-1~deb11u2
* OS version Debian 12

# Checklist:

- [x] I have read the Contributing documentation available here: https://snipe-it.readme.io/docs/contributing-overview
- [x] I have formatted this PR according to the project guidelines: https://snipe-it.readme.io/docs/contributing-overview#pull-request-guidelines
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/Presenters/AssetPresenter.php [hunk 1]
```diff
@@ -562,6 +562,36 @@ public function dynamicWarrantyUrl()
         return $url;
     }
 
+    /**
+     * Used to take user created Support URL and dynamically fill in the needed values per asset
+     * @return string
+     */
+    public function dynamicSupportUrl()
+    {
```
Comment (snipe) [inline app/Presenters/AssetPresenter.php]: I think we can probably combine these two, no? It's effectively the same code repeated.

**Review by snipe — CHANGES_REQUESTED** [review]
I think there's some duplication we can clean up in this.

**Comment by DrekiDegga** [comment]
@snipe  Sorry about that. I believe I've made the requested changes by merging the Dynamic URL Functions in AssetPresenter.php.

**Comment by snipe** [comment]
This looks a lot better, thank you. But c an you retarget to `develop`, per the developer docs?

**Pull request closed without merging**